src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -2100,6 +2100,32 @@ theorem toNat_abs {x : BitVec w} : x.abs.toNat = if x.msb then 2^w - x.toNat els
     rw [Nat.mod_eq_of_lt (by omega)]
   · simp [h]
 
+theorem getLsbD_abs {i : Nat} {x : BitVec w} :
+   getLsbD x.abs i = if x.msb then getLsbD (-x) i else getLsbD x i := by
+  by_cases h : x.msb <;> simp [BitVec.abs, h]
+
+theorem getMsbD_abs {i : Nat} {x : BitVec w} :
+    getMsbD (x.abs) i = if x.msb then getMsbD (-x) i else getMsbD x i := by
+  by_cases h : x.msb <;> simp [BitVec.abs, h]
+
+theorem msb_abs {w : Nat} {x : BitVec w} :
+    (x.abs).msb = if x.msb then (-x).msb else x.msb := by
```

Comment:
```suggestion
    (x.abs).msb = false := by
  simp only [BitVec.abs, neg_eq]
  have : (-x).msb = !x.msb := sorry
  cases h : x.msb
  <;> simp [h, this]
```

The absolute value is always positive, and thus the sign-bit (msb) is always false!

Reply:
oh right! thanks!

Reply:
Sorry, turns out I was wrong: the absolute value of `intMin is `intMin` itself, and thus still negative.
I edited the suggested statement to the actually correct version: the msb of the absolute value is always false, *unless* `x` is `intMin`.